filetablemodel.cpp

```diff
@@ -106,10 +107,23 @@ QVariant FileTableModel::data(const QModelIndex &index,
                               int role) const
 {
     // We validate arguments.
-    if (!index.isValid() || role != Qt::DisplayRole ||
-            index.column() >= columnCount() || index.row() >= rowCount())
+    if (!index.isValid() || index.column() >= columnCount() || index.row() >= rowCount())
+        return QVariant();
+    switch (role) {
+    case Qt::DisplayRole:
+        return m_data.at(index.column()).at(index.row());
+        break;
+    case Qt::CheckStateRole:
+        if ((index.column() == 0) && (index.row() < m_checkedStates.count())) {
+            return m_checkedStates[index.row()];
+        }
+        else {
```

Comment:
I corrected the problem with conditions in parentheses. What do you want for simple condition like if (!index.isValid() && (index.column() < 9)). Do you want parentheses around !index.isValid() too ? Regarding the opening braces or if condition,  if you give me the choice, I'd choose the same as Aleksey 
if (xxx) {
}